x-pack/plugins/observability_solution/apm/common/waterfall/typings.ts

```diff
@@ -6,83 +6,61 @@
  */
 
 import { Exception } from '../../typings/es_schemas/raw/error_raw';
-import { EventOutcome } from '../../typings/es_schemas/raw/fields/event_outcome';
-import { SpanLink } from '../../typings/es_schemas/raw/fields/span_links';
-import { TimestampUs } from '../../typings/es_schemas/raw/fields/timestamp_us';
 import { AgentName } from '../../typings/es_schemas/ui/fields/agent';
 
 export interface WaterfallTransaction {
-  timestamp: TimestampUs;
-  trace: { id: string };
-  service: {
-    name: string;
-    environment?: string;
-  };
-  agent: { name: AgentName };
-  event?: { outcome?: EventOutcome };
-  parent?: { id?: string };
-  processor: { event: 'transaction' };
-  transaction: {
-    duration: { us: number };
-    id: string;
-    name: string;
-    type: string;
-    result?: string;
-  };
-  faas?: {
-    coldstart?: boolean;
-  };
-  span?: {
-    links?: SpanLink[];
-  };
+  'timestamp.us': number[];
+  'trace.id': string[];
+  'service.name': string[];
+  'service.environment'?: string[];
+  'agent.name': string[];
+  'event.outcome'?: string[];
+  'parent.id'?: string[];
+  'processor.event': ['transaction'];
+  'transaction.duration.us': number[];
+  'transaction.id': string[];
+  'transaction.name': string[];
+  'transaction.type': string[];
+  'transaction.result'?: string[];
+  'faas.coldstart'?: boolean[];
+  'span.links.span.id'?: string[];
+  'span.links.trace.id'?: string[];
```

Comment:
Q: Why is everything defined as arrays?

Reply:
the data returned by 'fields' is always wrapped in an array : https://www.elastic.co/guide/en/elasticsearch/reference/current/search-fields.html#search-fields-response

Reply:
I tend to agree with @crespocarlos here. IMHO the APIs must return the same payload whether we query for _source or fields.

Reply:
It shall be done.